src/upload.rs

```diff
@@ -316,9 +320,18 @@ fn complete_registry(opt: &PublishOpt) -> Result<Registry> {
     let pypirc = load_pypirc();
     let (registry_name, registry_url) = if let Some(repository_url) = opt.repository_url.as_deref()
     {
-        let name = match repository_url {
-            PublishOpt::DEFAULT_REPOSITORY_URL => Some("pypi"),
-            PublishOpt::TEST_REPOSITORY_URL => Some("testpypi"),
+        let normalized_url = PublishOpt::normalize_url(repository_url);
+        let name = match normalized_url {
+            normalized
+                if normalized == PublishOpt::normalize_url(PublishOpt::DEFAULT_REPOSITORY_URL) =>
```

Comment:
I'd just remove the end slash in constant value instead.

Reply:
Hi @messense  Thanks for reviewing. That's a solution, too.
Then, what if user input this: `"https://test.pypi.org/legacy/"`?

Reply:
Aha Understood! I got your point, thanks for simplify the logic. Updated the code.